src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -318,13 +315,48 @@ public void describe(ColibriConferenceIQ.ChannelBundle channelBundle)
     public JSONObject getDebugState()
     {
         JSONObject debugState = new JSONObject();
+        debugState.put("receiverVideoConstraints", receiverVideoConstraintsMap);
+        debugState.put("maxReceiverVideoConstraints", maxReceiverVideoConstraints);
         debugState.put("displayName", displayName);
         debugState.put("expired", expired);
         debugState.put("statsId", statsId);
 
         return debugState;
     }
 
+    /**
+     * Computes and sets the {@link #maxReceiverVideoConstraints} from the
+     * specified video constraints.
+     *
+     * @param newVideoConstraints the map of receiver endpoint id -> video
+     *                            constraints that specifies who (which receiver endpoint) is viewing what
+     *                            (as determined by the video constraints) from this endpoint (which is the
+     *                            sender).
+     */
+    private void setReceiverVideoConstraints(
+        ImmutableMap<String, VideoConstraints> newVideoConstraints)
+    {
+        VideoConstraints oldReceiverMaxVideoConstraints
+            = this.maxReceiverVideoConstraints;
+
+        VideoConstraints defaultVideoConstraints
+            = new VideoConstraints(BitrateControllerConfig.Config.thumbnailMaxHeightPx());
+
+        VideoConstraints newReceiverMaxVideoConstraints = newVideoConstraints
+            .values()
+            .stream()
+            .max(Comparator.comparingInt(VideoConstraints::getIdealHeight))
+            .orElse(defaultVideoConstraints);
+
+        // for intra-bridge traffic we only care about the ideal height.
```

Comment:
I don't get why this commented refers to 'intra-bridge traffic', because `onMaxReceiverVideoConstraintsChanged` doesn't say anything about being specific to only intra-bridge traffic cases?